src/function/audio.rs

```diff
@@ -0,0 +1,296 @@
+    ///
+    /// The audio channel mask is a bit mask of the audio channels. The mask is a 32-bit integer with each bit representing a channel. The least significant bit is channel 1. The mask is used to specify the audio channels that are present in the audio stream. For example, a stereo stream would have a mask of 0x3 (channel 1 and channel 2).
+    pub channel_mask: Option<u32>,
+    /// Audio sample rate (Hz)
+    pub sample_rate: Option<u32>,
+    /// Audio sample size (bytes) so 2 bytes per sample (16 bit) would be 2.
+    pub sample_size: Option<u32>,
+    /// Audio sync type (capture only)
+    pub sync_type: Option<u32>,
+    /// Capture bInterval for HS/SS (1-4: fixed, 0: auto)
+    pub hs_interval: Option<u8>,
+    /// If channel has mute
+    pub mute_present: Option<bool>,
+    /// Terminal type
+    pub terminal_type: Option<u8>,
+    /// If channel has volume
+    pub volume_present: Option<bool>,
+    /// Minimum volume (in 1/256 dB)
+    pub volume_min: Option<i16>,
+    /// Maximum volume (in 1/256 dB)
+    pub volume_max: Option<i16>,
+    /// Resolution of volume control (in 1/256 dB)
+    pub volume_resolution: Option<i16>,
+    /// Name of the volume control function
+    pub volume_name: Option<String>,
+    /// Name of the input terminal
+    pub input_terminal_name: Option<String>,
+    /// Name of the input terminal channel
+    pub input_terminal_channel_name: Option<String>,
+    /// Name of the output terminal
+    pub output_terminal_name: Option<String>,
+}
+
+/// Builder for USB audio class 2 (UAC2) function.
+///
+/// Set capture or playback channel_mask to 0 to disable the audio endpoint.
+#[derive(Debug, Clone, Default)]
+#[non_exhaustive]
+pub struct Uac2Builder {
+    /// Audio capture configuration.
+    pub capture: Uac2Config,
+    /// Audio playback configuration.
+    pub playback: Uac2Config,
+    /// Maximum extra bandwidth in async mode
+    pub fb_max: Option<u32>,
+    /// The number of pre-allocated request for both capture and playback
+    pub request_number: Option<u32>,
+    /// The name of the interface
+    pub function_name: Option<String>,
+    /// Topology control name
+    pub control_name: Option<String>,
+    /// The name of the input clock source
+    pub clock_source_in_name: Option<String>,
+    /// The name of the output clock source
+    pub clock_source_out_name: Option<String>,
+}
+
+impl Uac2Builder {
+    /// Build the USB function.
+    ///
+    /// The returned handle must be added to a USB gadget configuration.
+    pub fn build(self) -> (Uac2, Handle) {
+        let dir = FunctionDir::new();
+        (Uac2 { dir: dir.clone() }, Handle::new(Uac2Function { builder: self, dir }))
+    }
+
+    /// Add audio capture configuration.
+    pub fn add_capture_config(&mut self, capture: Uac2Config) {
+        self.capture = capture;
+    }
+
+    /// Add audio playback configuration.
+    pub fn add_playback_config(&mut self, playback: Uac2Config) {
```

Comment:
I would remove this, since it provide no additional value over setting the field directly.